### ISSUE TITLE
Backport: Rancher 2.2.1 kube-apiserver in loop missing stat /etc/kubernetes/kube-api-authn-webhook.yaml: no such file or directory

### ISSUE THREAD

**Issue by alena1108** [rancher/rancher#19515]
Investigate and backport https://github.com/rancher/rancher/issues/19438

**Comment by alena1108** [comment]
Can be tested with `rancher/rancher:v2.2.3-rc2`. @superseb I know this bug wasn't easy to repro, and you've used some script to emulate the scenario. Can you share it with QA?

**Comment by superseb** [comment]
The issue is surfaced when the process in the container takes longer than a second to complete, where it will be removed before completing. I've tried reproducing it by delaying the container startup, but it actually has to start the container and then take longer than a second. So I made a script which pulls the given rke-tools version and replaces `bash`/`sh` with a sleep wrapper, making every command sleep for configurable seconds before executing the command.

To reproduce:

- Run `rancher/rancher:v2.2.2` and create a custom cluster
- Create a Linux node and install Docker to be added to the custom cluster
- Before adding the node, run the following script to preload the rke-tools image and add the sleep (`v0.1.27` is the rke-tools version used for the default k8s version for custom clusters, `5` is the amount of seconds to sleep before executing the actual command)

```
curl -sL https://gist.githubusercontent.com/superseb/10ddd7dceb1728dcfe6aaa402a2b4b71/raw/f0e35a400329110312e56b91a30af98f3e864829/rke-tools-slow.sh | bash -s -- v0.1.27 5
```

- Add the node to the custom cluster by running the command given in the UI
- Healthcheck for kube-apiserver should fail:

```
Failed to verify healthcheck: Failed to check https://localhost:6443/healthz for service [kube-apiserver] on host [x]: Get https://localhost:6443/healthz: dial tcp 127.0.0.1:6443: connect: connection refused, log: invalid authentication config: stat /etc/kubernetes/kube-api-authn-webhook.yaml: no such file or directory
```

Running the same sequence when using `rancher/rancher:v2.2.3-rc2`, should give a working cluster.
